gateway/gateway.go

```diff
@@ -5,33 +5,39 @@ import (
 	"net/http"
 	"sort"
 
-	coreiface "github.com/ipfs/interface-go-ipfs-core"
-	path "github.com/ipfs/interface-go-ipfs-core/path"
+	cid "github.com/ipfs/go-cid"
+	"github.com/ipfs/go-libipfs/blocks"
+	"github.com/ipfs/go-libipfs/files"
+	iface "github.com/ipfs/interface-go-ipfs-core"
+	"github.com/ipfs/interface-go-ipfs-core/path"
 )
 
-// Config is the configuration that will be applied when creating a new gateway
-// handler.
+// Config is the configuration used when creating a new gateway handler.
 type Config struct {
-	Headers  map[string][]string
-	Writable bool
+	Headers map[string][]string
 }
 
-// NodeAPI defines the minimal set of API services required by a gateway handler
-type NodeAPI interface {
-	// Unixfs returns an implementation of Unixfs API
-	Unixfs() coreiface.UnixfsAPI
+// API defines the minimal set of API services required for a gateway handler.
+type API interface {
```

Comment:
nit: @hacdias something like `IPFSBackend` would be more descriptive, but can be changed later, won't block on this one thing  (this interface will most likely get revamp in near future anyway)